detection_rules/rule.py

```diff
@@ -215,20 +215,29 @@ class FlatThreatMapping(MarshmallowDataclassMixin):
     sub_technique_ids: List[str]
 
 
+@dataclass(frozen=True)
+class AlertSuppressionDuration:
+    """Mapping to alert suppression duration."""
+    unit: definitions.TimeUnits
+    value: definitions.AlertSuppressionValue
+
+
 @dataclass(frozen=True)
 class AlertSuppressionMapping(MarshmallowDataclassMixin, StackCompatMixin):
     """Mapping to alert suppression."""
-    @dataclass
-    class AlertSuppressionDuration:
-        """Mapping to allert suppression duration."""
-        unit: definitions.TimeUnits
-        value: int
 
-    group_by: List[definitions.NonEmptyStr]
+    group_by: definitions.AlertSuppressionGroupBy
     duration: Optional[AlertSuppressionDuration]
     missing_fields_strategy: definitions.AlertSuppressionMissing
 
 
+@dataclass(frozen=True)
+class ThresholdAlertSuppression:
```

Comment:
If you move this up and rename it `BaseAlertSuppression`, you can inherit it into `AlertSuppressionMapping`. This also changes the name to be more generic in case future rule types also _only_ use this field

Reply:
I thought about that. I just made it align to the Kibana implementation.